### ISSUE TITLE
Onnx Export Script

### ISSUE THREAD

**Issue by thewh1teagle** [yukiarimo/aiflow#5]
Hey! could you please provide the onnx export script for the hanasu model?
also, is there any samples / demo I can see? I want to make sure it's higher quality than piper at least or StyleTTS2 before I'm training

**Comment by yukiarimo** [comment]
ONNX export function is in the models as a function. I’ll update it along side with preprocessing for a faster inference.

For the training, no samples cause people will steal our voices, but architecture is adopted from VITS-2 for 48 kHz.

I’ll tweak the convolutions to see how to improve the checkerboard, otherwise, if your dataset is studio quality and you haven’t used ASR, it will be fine

**Comment by thewh1teagle** [comment]
I'm experimenting with the model here

- https://github.com/thewh1teagle/hanasu

Currently I get huge loss and Nan values on the first steps... 
The first commit is just your code with cleaned-up tokenization (just moved tokenization/mel compute outside of the training code)

My dataset is studio quality + hand annotated phonemes of 20H

**Comment by yukiarimo** [comment]
Whoa! Okay, thanks for letting me know! For sure, we will look into that!

**Comment by yukiarimo** [comment]
Hello again. I've misread your comment. Yes, that implementation is different and will not work. I'll update the code soon. Feel free to use this repo. It is the most stable and supported version with lots of optimizations. Adding CoreML soon, too!

**Comment by yukiarimo** [comment]
Updated. Check out hanasu colab notebook

**Comment by thewh1teagle** [comment]
Thanks @yukiarimo I will try it soon. do you have any checkpoints I can start with to fine tune and save some training time?

**Comment by yukiarimo** [comment]
No. I trained exclusively on my and my VA’s voices, so I cannot provide checkpoints, unfortunately!

Start from scratch, after 8,000 steps (that’s around 30 minutes; approximately ~10 epochs, if I’m not wrong), you’ll hear a voice forming.

It’s so fun to watch it happen live!

**Comment by thewh1teagle** [comment]
Gotcha, I'll train it soon. What's the RTF on cpu? Also is 20-40h dataset sufficient?

**Comment by yukiarimo** [comment]
1. It depends on the CPU. But on my MacBook M1 with 16 GB of RAM, MPS inference is around 40 minutes of inference for 7 hours of audio.
2. Yes, it's 100% enough!

> Note: if you are planning to generate audiobooks with it, do not pass more than 70k characters into one inference call, otherwise RAM/VRAM usage will explode. We are still investigating this!
